test/jdk/java/awt/print/PrinterJob/PrintImage.java

```diff
@@ -21,276 +21,167 @@
-                Object target = e.getSource();
-                if( target.equals(print1Menu) ) {
-                        printMain1();
-                }
-                else if( target.equals(print2Menu) ) {
-                        printMain2();
-                }
-                else if( target.equals(exitMenu) ) {
-                        dispose();
-                }
+    public void actionPerformed(ActionEvent e) {
+        Object target = e.getSource();
+        if (target.equals(print1Menu)) {
+            printMain1();
+        } else if (target.equals(print2Menu)) {
+            printMain2();
+        } else if (target.equals(exitMenu)) {
+            dispose();
         }
+    }
 
-        private void printMain1(){
+    private void printMain1() {
 
-                PrinterJob printerJob = PrinterJob.getPrinterJob();
-                PageFormat pageFormat = printerJob.defaultPage();
+        PrinterJob printerJob = PrinterJob.getPrinterJob();
+        PageFormat pageFormat = printerJob.defaultPage();
 
-                printerJob.setPrintable((Printable)printImageCanvas, pageFormat);
+        printerJob.setPrintable(printImageCanvas, pageFormat);
 
-                if(printerJob.printDialog()){
-                        try {
-                                printerJob.print();
-                        }
-                        catch(PrinterException p){
-                        }
-                }
-                else
-                        printerJob.cancel();
-        }
+        if (printerJob.printDialog()) {
+            try {
+                printerJob.print();
+            } catch (PrinterException e) {
+                e.printStackTrace();
+            }
+        } else
+            printerJob.cancel();
+    }
 
-        private void printMain2(){
+    private void printMain2() {
 
-                PrinterJob printerJob = PrinterJob.getPrinterJob();
-                PageFormat pageFormat = printerJob.pageDialog(printerJob.defaultPage());
+        PrinterJob printerJob = PrinterJob.getPrinterJob();
+        PageFormat pageFormat = printerJob.pageDialog(printerJob.defaultPage());
 
-                printerJob.setPrintable((Printable)printImageCanvas, pageFormat);
-
-                if(printerJob.printDialog()){
-                        try {
-                                printerJob.print();
-                        }
-                        catch(PrinterException p){
-                        }
-                }
-                else
-                        printerJob.cancel();
-        }
+        printerJob.setPrintable(printImageCanvas, pageFormat);
 
+        if (printerJob.printDialog()) {
+            try {
+                printerJob.print();
+            } catch (PrinterException e) {
+                e.printStackTrace();
+            }
+        } else
+            printerJob.cancel();
+    }
 }
 
 class PrintImageCanvas extends Canvas implements Printable {
```

Comment:
I propose making the `PrintImageCanvas` class a nested (`private static`) class inside `PrintImage`.